vavr/src/main/java/io/vavr/control/Option.java

```diff
@@ -177,7 +177,7 @@
      */
     default <R> Option<R> collect(PartialFunction<? super T, ? extends R> partialFunction) {
         Objects.requireNonNull(partialFunction, "partialFunction is null");
-        return filter(partialFunction::isDefinedAt).map(partialFunction::apply);
+        return flatMap(partialFunction.lift()::apply);
```

Comment:
@skestle Just recognized that the `::apply` is not necessary. Does it compile if you leave it away? Could you please check that, I'm on vacation and have no IDE at hand...
Thx!